### ISSUE TITLE
Highlight other code block keywords on hovering over their `end`

### ISSUE THREAD

**Issue by Seelengrab** [julia-vscode/julia-vscode#1269]
A friend of mine recently asked about whether `if`, `elseif` and `else` can be highlighted based on whether you're hovering over `end` or not. This could potentially also be done with other blocks, such as `for`, `while`, `begin`.. maybe more?

I'm not sure how complicated this is, but in C# with Visual Studio (non-code) something similar exists and as far as I know, they're using their language server as well, so it should be doable.

**Comment by pdimens** [comment]
It's not what you're describing per se, but vscode has a useful addon called [Indenticator](https://github.com/SirTori/indenticator) that highlights the indent guide for that indentation level. It's a decent proxy for the feature you're describing if your code is tab-formatted after the `if`/`while`/`for` etc. blocks.
![image](https://user-images.githubusercontent.com/19176506/83560546-0a40ca00-a4dc-11ea-96a1-e2a2984ab35c.png)

**Comment by Seelengrab** [comment]
Yes, but this only works if the blocks are indented. The line is also quite thin, so highlighting the keywords in C# helps him read the code easier, especially in dense if..else if .. else if .. else constructs where the indentation line doesn't exist because of the style he's using.
